### PR TITLE
Edit token acls

### PR DESCRIPTION
**Pull request by mcoughlin**
This PR edits  token acls.

### CODE REVIEW

File: skyportal/handlers/api/internal/token.py [hunk 1]
```diff
@@ -135,6 +137,118 @@ def get(self, token_id=None):
             data = session.scalars(stmt).all()
             return self.success(data=data)
 
+    @auth_or_token
+    def put(self, token_id):
+        """
+        ---
+        description: Update token
+        tags:
+          - telescopes
```
Comment (guynir42) [inline skyportal/handlers/api/internal/token.py:146]:
```suggestion
          - tokens
```

File: skyportal/handlers/api/internal/token.py [hunk 2]
```diff
@@ -135,6 +137,118 @@ def get(self, token_id=None):
...
+                token = session.scalars(
+                    Token.select(session.user_or_token, mode="update").where(
+                        Token.id == token_id
+                    )
+                ).first()
+                if token is None:
+                    return self.error(
+                        'Either the specified token does not exist, '
+                        'or the user does not have the necessary '
+                        'permissions to delete it.'
```
Comment (guynir42) [inline skyportal/handlers/api/internal/token.py:179]:
```suggestion
                        'Either the specified token does not exist, '
                        'or the user does not have the necessary '
                        'permissions to update it.'
```

File: skyportal/handlers/api/internal/token.py [hunk 3]
```diff
@@ -135,6 +137,118 @@ def get(self, token_id=None):
...
+                                ACL.select(session.user_or_token).where(
+                                    ACL.id == acl_id
+                                )
+                            ).first()
+                            is not None
+                            for acl_id in new_acl_ids
+                        ]
+                    ):
+                        return self.error(
+                            "Improperly formatted parameter aclIds; must be an array of strings."
```
Comment (guynir42) [inline skyportal/handlers/api/internal/token.py:225]: It seems like you are not just testing that they are strings, you are also checking they are legit ACLs. Maybe the error message should reflect that.

**Review by guynir42 — APPROVED** [review]
Very cool. I have some minor comments.

File: skyportal/handlers/api/internal/token.py (hunk 3 above)
Comment (mcoughlin) [reply skyportal/handlers/api/internal/token.py]: Good point

**Pull request merged**